src/Microsoft.DotNet.Build.Tasks/Targets/packageresolve.targets

```diff
@@ -34,23 +42,73 @@
       <Output TaskParameter="ResolvedReferences" ItemName="Reference" />
       <Output TaskParameter="ResolvedCopyLocal" ItemName="None" />
     </ResolveNuGetPackages>
+
+    <PropertyGroup>
+      <!-- Temporary workaround -->
+      <NugetTargetFrameworkMoniker Condition="'$(NugetTargetFrameworkMoniker)' == '.NETPortable,Version=v4.5,Profile=Profile7'">DNXCore,Version=v5.0</NugetTargetFrameworkMoniker>
+    </PropertyGroup>
+    
+    <!-- Use new task for project.json -->
+    <ResolveNuGetPackageAssets Condition="'$(ProjectJson)' != ''"
+                               Architecture="$(PlatformTarget)"
+                               Configuration="$(Configuration)"
+                               Language="$(Language)"
+                               IngoreLockFile="$(NuGetIngoreLockFile)"
+                               PackageRoot="$(PackagesDir)"
+                               ProjectFile="$(MSBuildProjectFullPath)"
+                               TargetFrameworkMonikers="$(NugetTargetFrameworkMoniker)"
+                               TargetPlatformMonikers="$(TargetPlatformMoniker)"
+                               TransitiveProjectReferences="">
+      <Output TaskParameter="ResolvedAnalyzers" ItemName="Analyzer" />
+      <Output TaskParameter="ResolvedReferences" ItemName="Reference" />
+      <Output TaskParameter="ResolvedCopyLocalItems" ItemName="None" />
+    </ResolveNuGetPackageAssets>
+
+    <!-- We may have an indirect package reference that we want to replace with a project reference -->
+    <ItemGroup>
+      <!-- Convert to filenames so that we can intersect -->
+      <_ProjectReferenceFilenames Include="@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_ProjectReferenceFilenames>
+
+      <_ReferencesFileNames Include="@(Reference->'%(FileName)%(Extension)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_ReferencesFileNames>
+
+      <_NoneFileNames Include="@(None->'%(FileName)%(Extension)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_NoneFileNames>
+      
+      <!-- Intersect project-refs with package-refs -->
+      <_ReferenceFileNamesToRemove Include="@(_ReferencesFileNames->'%(OriginalIdentity)')" Condition="'@(_ProjectReferenceFilenames)' == '@(_ReferencesFileNames)' and '%(Identity)' != ''"/>
+      <_NoneFileNamesToRemove Include="@(_NoneFileNames->'%(OriginalIdentity)')" Condition="'@(_ProjectReferenceFilenames)' == '@(_NoneFileNames)' and '%(Identity)' != ''"/>
+
+      <Reference Remove="@(_ReferenceFileNamesToRemove)" />
+      <None Remove="@(_NoneFileNamesToRemove)"/>
+    </ItemGroup>
+    
+    <Message Text="Excluding @(_ReferenceFileNamesToRemove);@(_NoneFileNamesToRemove) from package references since the same file is provided by a project refrence."
+             Condition="'@(_ReferenceFileNamesToRemove)' != '' AND '@(_NoneFileNamesToRemove)' != ''"/>
```

Comment:
I assume you wanted this condition to be an OR alternatively you could concat like '@(_ReferenceFileNamesToRemove)@(NoneFileNamesToRemove)' != ''"